src/mbgl/style/style_impl.cpp

```diff
@@ -309,6 +309,9 @@ void Style::Impl::onSpriteError(std::exception_ptr error) {
     lastError = error;
     Log::Error(Event::Style, "Failed to load sprite: %s", util::toString(error).c_str());
     observer->onResourceError(error);
+    // Unblock rendering tiles (even though sprite request has failed).
+    spriteLoaded = true;
+    observer->onUpdate();
```

Comment:
This will only unblock rendering in continuous mode. Still image mode will still fail because the `onResourceError` terminates the whole map render request.